docs/excel/excel-add-ins-ranges-unbounded.md

```diff
@@ -0,0 +1,36 @@
+---
+title: Read or write to an unbounded range using the Excel JavaScript API
+description: 'Learn how to use the Excel JavaScript API to read or write to an unbounded range.'
+ms.date: 04/05/2021
+localization_priority: Normal
+---
+
+# Read or write to an unbounded range using the Excel JavaScript API
+
+This article describes how to read and write to an unbounded range with the Excel JavaScript API. For the complete list of properties and methods that the `Range` object supports, see the [Excel.Range class](/javascript/api/excel/excel.range).
+
+An unbounded range address is a range address that specifies either entire columns or entire rows. For example:
+
+- Range addresses comprised of entire columns:<ul><li>`C:C`</li><li>`A:F`</li></ul>
+- Range addresses comprised of entire rows:<ul><li>`2:2`</li><li>`1:4`</li></ul>
+
+## Read an unbounded range
+
+When the API makes a request to retrieve an unbounded range (for example, `getRange('C:C')`), the response will contain `null` values for cell-level properties such as `values`, `text`, `numberFormat`, and `formula`. Other properties of the range, such as `address` and `cellCount`, will contain valid values for the unbounded range.
+
+## Write to an unbounded range
+
+You cannot set cell-level properties such as `values`, `numberFormat`, and `formula` on an unbounded range because the input request is too large. For example, the following code snippet is not valid because it attempts to specify `values` for an unbounded range. The API returns an error if you attempt to set cell-level properties for an unbounded range.
```

Comment:
```suggestion
You cannot set cell-level properties such as `values`, `numberFormat`, and `formula` on an unbounded range because the input request is too large. For example, the following code example is not valid because it attempts to specify `values` for an unbounded range. The API returns an error if you attempt to set cell-level properties for an unbounded range.
```